Service/Pdf/PdfGenerator.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EMS\CommonBundle\Service\Pdf;
+
+use Symfony\Component\DomCrawler\Crawler;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+
+class PdfGenerator
+{
+    /** @var PdfPrinterInterface */
+    private $pdfPrinter;
+
+    public function __construct(PdfPrinterInterface $pdfPrinter)
+    {
+        $this->pdfPrinter = $pdfPrinter;
+    }
+
+    public function getResponse(string $html): StreamedResponse
+    {
+        $metaTags = $this->getMetaTags($html);
+        $pdfPrintOptions = new PdfPrintOptions($this->sanitizeMetaTags($metaTags));
+        $filename = $metaTags[PdfInterface::FILENAME] ?? 'export.pdf';
+
+        return $this->pdfPrinter->getStreamResponse(new Pdf($filename, $html), $pdfPrintOptions);
+    }
+
+    private function getMetaTags(string $html): array
```

Comment:
Should be part of an HtmlParser class (S-OLID)

Reply:
Added HtmlCrawler but still need a getMetaTags function :) Logic is specific for PdfGenerator